src/interface/viewport.c

```diff
@@ -1537,6 +1539,1056 @@ void viewport_park_entrance_paint_setup(uint8 direction, int height, rct_map_ele
+		viewport_surface_smooth_north_edge();
+		viewport_surface_smooth_south_edge();
+		viewport_surface_smooth_east_edge();
+	}
+
+	if (!(RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_VIEWPORT_FLAGS, uint16) & VIEWPORT_FLAG_UNDERGROUND_INSIDE)) {
+		// loc_66122C:
+		uint8 al_edgeStyle = mapElement->properties.surface.slope & 0xE0;
+		uint8 di_type = mapElement->type & 0x80;
+
+		uint32 eax = al_edgeStyle + di_type * 2;
+		if (eax != 32 && eax != 0 && eax != 96 && eax != 64) {
+			log_info("eax: %d", eax);
+		}
+
+		viewport_surface_paint_struct_0 from = RCT2_GLOBAL(0x0097B5C0 + eax, viewport_surface_paint_struct_0);
+		RCT2_GLOBAL(0x009E329C, uint32) = from.var_00;
+		RCT2_GLOBAL(0x009E32A0, uint32) = from.var_04;
+		RCT2_GLOBAL(0x009E32A4, uint32) = from.var_08;
+		RCT2_GLOBAL(0x009E32A8, uint32) = from.var_0C;
+		RCT2_GLOBAL(0x009E32AC, uint32) = from.var_10;
+
+		for (int i = 0; i <= 0x7C; i += 4) {
+			RCT2_GLOBAL(0x009E2F30 + i, uint32) = RCT2_GLOBAL(0x009E3138 + i, uint32);
+			RCT2_GLOBAL(0x009E2EAE + i, uint32) = RCT2_GLOBAL(0x009E30B6 + i, uint32);
+		}
+
+		RCT2_CALLPROC_X(0x65F63B, 0x99999999, 0xAAAAAAAA, 0xBBBBBBBB, height / 16, 0xDDDDDDDD, 0, 0);
+		RCT2_CALLPROC_X(0x65F77D, 0x99999999, 0xAAAAAAAA, 0xBBBBBBBB, height / 16, 0xDDDDDDDD, 0, 0);
+		RCT2_CALLPROC_X(0x65EB7D, 0x99999999, 0xAAAAAAAA, 0xBBBBBBBB, height / 16, 0xDDDDDDDD, 0, 0);
+		RCT2_CALLPROC_X(0x65F0D8, 0x99999999, 0xAAAAAAAA, 0xBBBBBBBB, height / 16, 0xDDDDDDDD, 0, 0);
+	}
+
+	if ((RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_VIEWPORT_FLAGS, uint16) & VIEWPORT_FLAG_HIDE_VERTICAL) == 0) {
+		// loc_66122C:
+	}
+
+	RCT2_GLOBAL(0x009E3298, uint16) = 0;
+	if (mapElement->properties.surface.terrain & 0x1F) {
+		// loc_6615A9: (water height)
+		RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8) = VIEWPORT_INTERACTION_ITEM_WATER;
+
+		uint16 ax = height + 16;
+		uint16 dx = (mapElement->properties.surface.terrain & 0x1F) << 4;
+
+		RCT2_GLOBAL(0x009E3298, uint16) = dx;
+		if ((RCT2_GLOBAL(0x9DEA6F, uint8_t) & 1) == 0) {
+			RCT2_GLOBAL(0x0141E9DC, uint16) = dx;
+
+			int image_offset = 0;
+			if (dx > ax) {
+				image_offset = byte_97B740[ebx & 0xF];
+			}
+
+			int image_id = image_offset + 0x610013B8;
+			sub_98196C(image_id, 0, 0, 32, 32, -1, dx, get_current_rotation());
+
+			sub_68818E(5053 + image_offset, 0, 0);
+
+			for (int i = 0; i <= 0x7C; i += 4) {
+				RCT2_GLOBAL(0x009E3138 + i, uint32) = RCT2_GLOBAL(0x009E2F30 + i, uint32);
+				RCT2_GLOBAL(0x009E30B6 + i, uint32) = RCT2_GLOBAL(0x009E2EAE + i, uint32);
+			}
+
+			RCT2_CALLPROC_X(0x66039B, 0x99999999, 0, 0xBBBBBBBB, dx / 16, 0xDDDDDDDD, 0, 0);
+			RCT2_CALLPROC_X(0x6604F1, 0x99999999, 0, 0xBBBBBBBB, dx / 16, 0xDDDDDDDD, 0, 0);
+			RCT2_CALLPROC_X(0x65F8B9, 0x99999999, 0, 0xBBBBBBBB, dx / 16, 0xDDDDDDDD, 0, 0);
+			RCT2_CALLPROC_X(0x65FE26, 0x99999999, 0, 0xBBBBBBBB, dx / 16, 0xDDDDDDDD, 0, 0);
+		}
+	}
+
+	if (mapElement->properties.surface.ownership & 0x0F
+		&& !(RCT2_GLOBAL(0x009DEA6F, uint8) & 1)) {
+		// loc_66195F:
```

Comment:
I think this block is drawing the park fences, perhaps extract to another function?